
# Mathematik für Biologiestudierende

Wintersemester 2024/25

27.11.2024

&copy; 2024 Prof. Dr. Rüdiger W. Braun 

# Schließende Statistik

# Hypothesentests

## Beispiel

Es wird vermutet, dass Wärme den Keimerfolg eines Getreides verringert.  900 Saatkörner werden ausgebracht.

* bunte Felder:  Erfolg
* schwarze Felder:  Misserfolg

<img src="bilder/saatgut.svg" alt="Keimerfolge in kühlem und warmem Klima" width="50%"/>

## Frage

Gibt es weniger Erfolge bei dem Anbauversuch im Warmen?

* im kühlem Klima gab es 720 Keimerfolge
* im warmen nur 696

die richtige Frage ist aber

* beruht der Unterschied auf Zufall
* oder ist er signifikant?

## Zwei Probleme

die aber miteinander verbunden sind

### Zufallseinflüsse unvermeidlich

* Sichere Aussagen unmöglich
* Vorgabe des Signifikanzniveaus begrenzt Fehlerwarscheinlichkeit

### Binäre Antwort wird erwartet

obwohl prinzipiell drei Szenarien möglich sind

* klare Hinweise auf "ja"
* klare Hinweise auf "nein"
* unklares Ergebnis

"klar" und "unklar" bemisst sich nach dem Signifikanzniveau

## Beispiel Saatgut

Generell sind vier Ausgänge des Experiments möglich

#### Beeinträchtigt die Temperaturerhöhung den Keimerfolg?

* Die Temperaturerhöhung beeinträchtigt den Keimerfolg nicht und die statistische Auswertung des Experiments führt zur Antwort "nein"

  &#128994; Korrekte Antwort

*  Die Temperaturerhöhung beeinträchtigt den Keimerfolg nicht und die statistische Auswertung des Experiments führt zur Antwort "ja"

    &#128308; Falsche Antwort

* Die Temperaturerhöhung beeinträchtigt den Keimerfolg und die statistische Auswertung des Experiments führt zur Antwort "nein"

  &#128308; Falsche Antwort

* Die Temperaturerhöhung beeinträchtigt den Keimerfolg und die statistische Auswertung des Experiments führt zur Antwort "ja"

  &#128994; Korrekte Antwort

Was soll im Fall unklarer Datenlage die Antwort sein?

* Forschungsteam hat Zusammenhang zwischen Temperatur und Keimerfolg entdeckt und will Ergebnis in angesehener Zeitschrift publizieren
* Die Zeitschrift fordert stichhaltige Beweise
* Wenn die Zeitschrift die Datenlage für unklar hält, wird sie die Publikation ablehnen

Die Zeitschrift möchte ausschließen, dass ein zufälliger Effekt aufgebauscht wird

* *Nullhypothese*:  Es gibt keinen Effekt
* Das Gegenteil der Nullhypothese ist die *Alternative*
* Die Alternative ist also das, was "bewiesen" werden soll
* Wenn die Daten klar für die Alternative sprechen, nehme ich sie an
* In allen anderen Fällen behalte ich die Nullhypothese bei

* kniffelige Frage:  Was heißt hier "klar"

# Nullhypothese und Alternativhypothese

Durch die Auswahl der Stichprobe kommt Zufall ins Spiel. Falsche Antworten sind unvermeidbar.

* Ziel der Statistik ist es, Schranken für die Wahrscheinlichkeit falscher Antworten zu geben
* *Nullhypothese* $H_0$: Das ist diejenige Hypothese, deren fälschliche Ablehnung man nach Möglichkeit vermeiden will
* *Alternativhypothese* $H_1$: Das ist die Alternative zur Nullhypothese

**bei unklarer Datenlage wird also die Nullhypothese beibehalten**

* Wissenschaft ist konservativ.  Wer mit einer neuen Idee kommt, muss zeigen, dass sie besser ist als die alte
* typische Nullhypothesen:
    * bestehende Theorie so gut wie der neue Ansatz
    * der untersuchte Stoff ist ohne Einfluss
    * das Präparat ist wirkungslos
    * der beobachtete Unterschied im Gen ist folgenlos 

## Kleines Theaterstück

* Forscher:  Habe wichtige Beobachtung gemacht
* Statistikerin:  Ist doch alles Zufall
* Forscher:  Kann überhaupt nicht sein
* Statistikerin:  Zeigen Sie mal die Daten
* Forscher zeigt Daten
* Statistikerin rechnet


#### Happy End

* Statistikerin:  Mit Wahrscheinlichkeit 1% beruht das Ergebnis auf Zufall
* Forscher:  das ist eine sehr kleine Wahrscheinlichkeit
* Forscher reicht Arbeit bei Zeitschrift ein

#### offenes Ende

* Statistikerin:  Mit Wahrscheinlichkeit 20% beruht das Ergebnis auf Zufall
* Forscher:  Ich hätte gedacht, die Wahrscheinlichkeit sei viel kleiner
* Abgang Tür rechts zurück ins Labor

In Wahrheit sind Forscher und Statistikerin meist dieselbe Person

# Fehler erster und zweiter Art

* *Der Fehler 1. Art* ist die fälschliche Ablehnung der Nullhypothese
* *Der Fehler 2. Art* ist die fälschliche Beibehaltung der Nullhypothese 

Die Priorität liegt auf der Vermeidung des Fehlers 1. Art.  Diese Asymmetrie ist ein entscheidendes Merkmal der Testtheorie.

|                     | $H_0$ wird beibehalten | $H_0$ wird abgelehnt  |
|---------------------|:----------------------:|:---------------------:|
| $H_0$ **trifft zu** | richtige Entscheidung  | Fehler 1. Art         |
| $H_1$ **trifft zu** | Fehler 2. Art          | richtige Entscheidung |

## Sprechweise

* $H_0$ wird beibehalten oder abgelehnt
* $H_1$ wird angenommen oder verworfen

## Signifikanztests

* Für den Fall, dass $H_0$ zutrifft, bezeichnet man die Wahrscheinlichkeit, dass $H_0$ trotzdem abgelehnt wird, als *Fehlerwahrscheinlichkeit erster Art*
* Ein Test heißt *Signifikanztest* zum Niveau $\alpha$, wenn alle Fehlerwahrscheinlichkeiten erster Art $\le \alpha$ sind
* Das übliche Niveau ist 0.05
* Für den Fall, dass $H_0$ nicht zutrifft, bezeichnet man die Wahrscheinlichkeit, dass $H_0$ trotzdem beibehalten wird, als *Fehlerwahrscheinlichkeit zweiter Art*

# Binomialtests

### Beispiel: Saatgut

* Wir konstruieren einen Test zum Signifikanzniveau $\alpha = 0.05$
* Stichprobenumfang ist 900
* $p_0 = 0.80$ ist die Vergleichswahrscheinlichkeit, denn in kühlem Klima keimen 80% des Saatguts
* $p$ ist die unbekannte tatsächliche Wahrscheinlichkeit, dass ein zufällig herausgegriffenes Korn keimt
* Nullhypothese $H_0 = \{ p \ge p_0 \}$, d.h. die Nullhypothese besagt, dass der Keimerfolg in warmem Klima *nicht* geringer ist als in kühlem
  

* Bei der Bestimmung des Fehlers 1. Art gehen wir davon aus, dass $H_0$ wahr ist
* `Python`sagt: Mit Wahrscheinlichkeit 0.0452 beobachten wir in diesem Fall 699 oder weniger Keimerfolge
* Wenn wir also sagen:  Bei  oder weniger Bakterien wird $H_0$ abgelehnt, dann machen wir den Fehler 1. Art mit
  einer Wahrscheinlichkeit von ca 4.5%

Wo kommen diese Zahlen her:

In [1]:
import numpy as np
np.set_printoptions(legacy='1.21')
from scipy import stats

In [2]:
P = stats.binom(900, 0.80)  #  W'keitsverteilung unter Nullhypothese

In [5]:
P.cdf(699)   # cumulative distribution function
# W'keit, dass 699 odfer weniger Keimerfolge

0.045195625799864816

Zum Vergleich:

In [6]:
P.cdf(700)

0.05346354525840081

Diese Fehlerwahrscheinlichkeit ist höher als das Signifikanzniveau $\alpha=0.05$

Wir hätte ich den Wert 700 finden können ausgehend von $\alpha=0.05$

In [7]:
P.ppf(0.05)  # percent point function

700.0

Das ist das kleinste $k$, dessen Wert über 0.05 liegt



## Entscheidungsregel:

* $n=900$ und $p_0=0.80$ und $H_0=\{p\ge p_0\}$ und $\alpha=0.05$

* Die Nullhypothese wird abgelehnt, wenn 699 oder weniger Erfolge beobachtet werden
* Bei 700 oder mehr Erfolgen wird die Nullhypothese beibehalten

Im Beispiel hatten wir 696 Keimerfolge.  Also ist nachgewiesen, dass der Keimerfolg in warmem Klima sinkt

# Fehlerwahrscheinlichkeit zweiter Art

* Wie groß ist die Fehlerwahrscheinlichkeit zweiter Art?

* Das ist keine gute Frage
* Wenn die Keimerfolgswahrscheinlichkeit von 80% auf 79.5% sinkt, dann ist das praktisch nicht nachzuweisen
* Sinnvoll ist folgende Frage

>  Angenommen, die Keimerfolgswahrscheinlichkiet sinkt auf 75%, mit welcher Wahrscheinlichkeit wird unser Test diesen Rückgang  entdecken?

* Wenn $q$ die Fehlerwahrscheinlichkeit zweiter Art ist, dann bezeichnet man $ 1 - q $ als *Power* des Tests
* Die Power hängt also davon ab, welche Annahme man über den Abstand zwischen Nullhypothese und Alternative macht

Fehlerwahrscheinlichkeit zweiter Art, wenn die Keimwahrscheinlichkeit auf 75% sinkt

* Entscheidungsregel: Bei 700 oder mehr Keimerfolgen wird $ H_0 $ beibehalten
* Wie wahrscheinlich ist dieses Ergebnis, wenn tatsächlich nur 75% aller Saatkörner keimen?
* Gesucht
$$   \sum_{k=700}^{900} B_{900,\,0.75}(k)
  = 1 - \sum_{k=0}^{699} B_{900,\,0.75}(k)
  = 1 - 0.9715 = 0.0285  $$

In [8]:
Q = stats.binom(900, 0.75)  #  W'keitsverteilung der Alternative
1 - Q.cdf(699)

0.02848516837071069

* Unter obiger Annahme beträgt die Fehlerwahrscheinlichkeit zweiter Art 2.85%
* Die Power beträgt entsprechend 97.15%% 
* Mit anderen Worten:  In 97\% der Fälle, bei denen die Keimerfolgswahrscheinlichkeit von 80% auf unter 75% sinkt, kann der Test diesen Fakt entdecken 

## Fehler erster und zweiter Art

<img src="bilder/fehler2.svg" alt="Fehler erster und zweiter Art" width="75%"/>

Die orangefarbenen Balken zeigen Fehlentscheidungen

## Kochrezepte 

## Ein- und zweiseitige Tests

* Ein ja/nein-Experiment mit unbekannter Erfolgswahrscheinlichkeit $p$ wird $n$-mal wiederholt
* Ziel:  Aussage über $p$ relativ zu einem Referenzwert $p_0$ 
* verschiedene Nullhypothesen sind denkbar
  * $H_0 : p \ge p_0$: einseitiger unterer Test
  * $H_0 : p \le p_0$: einseitiger oberer Test
  * $H_0 : p = p_0$: zweiseitiger Test
* Die Nullhypothese $H_0 \colon p \ne p_0$ macht keinen Sinn

## Einseitiger unterer Binomialtest zum Niveau  $\alpha$

* Gegeben sind unabhängige $B(1, p)$-verteilte Zufallsvariable $X_1, \dots, X_n$ mit unbekanntem $p$ sowie ein Signifikanzniveau $\alpha$
* Verglichen werden soll mit einem Referenzwert $p_0$
* Getestet wird die Nullhypothese $H_0 = \{p \ge p_0\}$ gegen die Alternative $H_1 = \{p < p_0\}$
* `P` ist die Binomialverteilung $B_{n,p_0}$
* Der Wert $c$ ist so zu wählen, dass `P.cdf(c-1)`$\le\alpha$ und `P.cdf(c)`$>\alpha$
* $c$ heißt *kritischer Wert*

### Entscheidungsregel:

* Die Nullhypothese wird abgelehnt, falls die Anzahl der Erfolge echt kleiner als $c$ ist
* Die Nullhypothese wird beibehalten, falls die Anzahl der Erfolge mindestens $c$ ist


## Beispiel zum unteren Binomialtest

* Im ungestörtem Boden sind 75% aller Bakterien L-Bakterien und 25% R-Bakterien
* Ein Pestizid wird verdächtigt, den Anteil an L-Bakterien zu senken
* Die Frage

> Sinkt der Anteil an L-Bakterien durch das Pestizid
  
  soll zum Signifikanzniveau $\alpha = 0.05$ beantwortet werden
  
* Dazu werden 27 Bakterien auf ihre Chiralität untersucht

In [9]:
P = stats.binom(27, 0.75)
alpha = 0.05
c = P.ppf(alpha)
c

16.0

Wenn 15 oder weniger L-Bakterien gefunden werden, dann lehnen wir die Nullhypothese ab

## Einseitiger oberer Binomialtest zum Niveau $\alpha$

* Gegeben sind unabhängige $B(1, p)$-verteilte Zufallsvariable $X_1, \dots, X_n$ mit unbekanntem $p$ sowie ein
Signifikanzniveau $\alpha$
* Verglichen werden soll mit einem Referenzwert $p_0$
* Getestet wird die Nullhypothese $H_0 = \{p \le p_0\}$ gegen die Alternative $H_1 = \{p > p_0\}$
* `P` ist die Binomialverteilung $B_{n,p_0}$
* Der kritische Wert $c$ ist so zu wählen, dass `P.cdf(c-1)`$<1-\alpha$ und `P.cdf(c)`$\ge1-\alpha$

### Entscheidungsregel:

* Die Nullhypothese wird abgelehnt, falls die Anzahl der Erfolge echt größer als $c$ ist
* Die Nullhypothese wird beibehalten, falls die Anzahl der Erfolge höchstens $c$ ist

## Beispiel zum oberen Binomialtest

* Zuchtlachsen wird Fischabfall und vegetarisches Futter zur Auswahl angboten
* 11% aller Lachse bevorzugen das Gemüse
* Der Geschmack der vegetarischen Kost wird durch Zusatz eines Aromastoffs verändet
* Die Frage

> Bevorzugen sie das aromatisierte Gemüse mit höherer Wahrscheinlichkeit als vorher?
  
  soll zum Signifikanzniveau $\alpha = 0.05$ beantwortet werden
  

* Einseitiger, oberer Binomialtest mit Nullhypothese $H_0 = \{p \le 0.11 \}$
* Stichprobenumfang $n = 38$

In [10]:
P = stats.binom(38, 0.11)
alpha = 0.05
P.ppf(1-alpha)

8.0

In [11]:
P.cdf(8)

0.9801967881175796

In [12]:
P.cdf(7)

0.9484147640607011

Also $c=8$

Wenn 9 oder mehr Fische das aromatisierte Futter bevorzugen, dann wird die Nullhypothese, dass das neue Futter nicht besser angenommen wird als das alte, abgelehnt

Wenn die Nullhypothese trotzdem gilt, dann machen wir mit Wahrscheinlichkeit

In [13]:
1 - P.cdf(8)

0.019803211882420402

den Fehler erster Art

# Beispiel Lachs: Power des Tests

* Was ist die Power des Tests, wenn tatsächlich 15% der Lachse die aromatisierte vegetarische Kost bevorzugen?
* $H_0$ wird abgelehnt bei mindestens 9 Erfolgen
* Im Beispiel $ p = 0.15 $

In [14]:
Q = stats.binom(38, 0.15)

* Wenn 8 oder weniger Erfolge beobachtet werden, mache ich den Fehler zweiter Art
* Also ist die Fehlerwahrscheinlichkeit zweiter Art gleich

In [15]:
Q.cdf(8)

0.8942848027814655

und die Power gleich

In [16]:
1 - Q.cdf(8)

0.1057151972185345

* Die Power beträgt nur 10.5%.  Das bedeutet zweierlei:
  * Der Test ist schlecht konzipiert:  Eine Verbesserung wird mit großer Wahrscheinlichkeit gar nicht entdeckt
  * Rückschlüsse auf die Nullhypothese sind nicht möglich

## Fehler erster und zweiter Art im Beispiel "Lachs"

<img src="bilder/powerlachs.svg" alt="Fehler erster und zweiter Art" width="75%"/>

Orangefarbene Balken zeigen Fehlentscheidungen

## Zusammenfassung

* Stichprobenumfang und Referenzwert 
* oberer, unterer oder zweiseitiger Test
* kritischer Wert
* Entscheidungsregel
* **dann das Experiment durchführen**
* Entscheidungsregel anwenden, um Nullhyposthese abzulehnen oder beizubehalten